apps/prairielearn/src/pages/instructorRequestCourse/instructorRequestCourse.html.ts

```diff
@@ -0,0 +1,230 @@
+import { html } from '@prairielearn/html';
+import { renderEjs } from '@prairielearn/html-ejs';
+import { CourseRequest } from '../../lib/db-types';
+
+export const RequestCourse = ({
+  course_requests,
+  resLocals,
+}: {
+  course_requests: CourseRequest[];
+  resLocals: Record<string, any>;
+}) => {
+  return html`
+    <!doctype html>
+    <html lang="en">
+      <head>
+        ${renderEjs(__filename, "<%- include('../partials/head')%>", {
+          ...resLocals,
+        })}
+        <script>
+          $(function () {
+            $('input[name=cr-role]').change(function () {
+              var role = this.value;
+              $('.question-form button').prop('disabled', role != 'instructor');
+              $('.role-comment').hide();
+              $('.role-comment-' + role).show();
+            });
+          });
+        </script>
+      </head>
+      <body>
+        ${renderEjs(__filename, "<%- include('../partials/navbar')%>", {
+          ...resLocals,
+          navPage: 'request_course',
+        })}
+        <main id="content" class="container">
+          ${course_requests.length > 0
+            ? html`
+                <!-- Previous Course Requests -->
+                <div class="card mb-4">
+                  <div class="card-header bg-primary text-white d-flex align-items-center">
+                    Course Requests
+                  </div>
+                  <div class="table-responsive">
+                    <table class="table table-sm table-hover table-striped">
+                      <thead>
+                        <tr>
+                          <th>Short Name</th>
+                          <th>Title</th>
+                          <th>Status</th>
+                        </tr>
```

Comment:
You removed the Details column here. It is not in the schema, but it is in the query:
https://github.com/PrairieLearn/PrairieLearn/blob/563c91c35c001618e096b363b66a969aa889dff4/apps/prairielearn/src/pages/instructorRequestCourse/instructorRequestCourse.sql#L4-L13

Reply:
Oh, good catch! I noticed details wasn't in the schema or database so I took it as a feature that never got developed. I didn't think about us manually generating it in the SQL. That seems like logic we should move to the html.ts, right?

Reply:
Indeed, you're right, this could be removed from the query and moved to the html.ts file, so a change in the schema may not be necessary.

Reply:
I made a pass at doing this. We'll see what the rest of the crew things about it.

Reply:
I didn't realize there was a join with the approver user. In that case, I think it makes sense to either keep the previous query computation, or at least add the approver name as a column to the query, then use that name in the computation. In that case, you can check what is done in:
https://github.com/PrairieLearn/PrairieLearn/blob/563c91c35c001618e096b363b66a969aa889dff4/apps/prairielearn/src/pages/instructorAssessmentManualGrading/instanceQuestion/instanceQuestion.html.ts#L10-L15

and create a `CourseRequestRowSchema` object that adds the new column.

Reply:
Yep, let's update the schema to avoid a nasty N+1 query here. The "new" style Matt has been pushing for would be something like this:

```ts
const CourseRequestRowSchema = z.object({
  course_request: CourseRequestSchema,
  approved_by_user: UserSchema,
});
```

Of course, update the query to match. `to_json` is your friend!

Reply:
Yes! Please switch to this, it's sooooo much cleaner+safer+easier!

Reply:
It's nice getting rid of the need for COALESCE in the SQL.

Reply:
Implemented in c1a0a26a